### ISSUE TITLE
Lava | Array Filters Have Unexpected Behaviors When Chained Together

### ISSUE THREAD

**Issue by jamrphy-bwd** [SparkDevNetwork/Rock#5846]
### Description

Lava Array Filters are inconsistent in handling `List`'s and `IEnumerable`'s, which will sometimes cause unexpected results.

### Actual Behavior

Consider the following lava, tested on DotLiquid v16.4 demo:
```liquid
{% assign funArray = '2,1,3,5,4,2' | Split:',' | Distinct %}
{{ funArray | Shuffle | ToJSON }}<br>
{{ funArray | Contains:'2' }}
```
This outputs, without variation:
```liquid
[ "2", "1", "3", "5", "4" ]
false
```

Remove the "Distinct" Filter from the assignment, and everything else works.
```liquid
{% assign funArray = '2,1,3,5,4,2' | Split:',' %}
```

### Expected Behavior

Output from the test above should vary the array in the `Shuffle` filter, at random, and return `true` for the `Contains` filter. 

In general, I'd expect the type of an "array" in lava to be a C# `List` (or for an `IEnumerable` to become a `List` after use in a filter).

### Steps to Reproduce

- Go to any ol' HTML block or your favorite Lava Tester page.
- Create an Array in Lava
- Use the created array in multiple "Array" filters from the [Documentation](https://community.rockrms.com/lava/filters/array-filters), such as {{ array | Distinct | Contains:'item' }}, where 'item' is actually in the array.
- See confusing results

### Issue Confirmation

- [X] Perform a search on the [Github Issues](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ARock) to see if your bug or enhancement is already reported.
- [X] Try to reproduce the problem on a fresh install or on the [demo site](https://rock.rocksolidchurchdemo.com/).

### Rock Version

16.4

### Client Culture Setting

en-US

**Comment by chead4** [comment]
@jamrphy-bwd After reviewing this issue, we've discovered that this issue is only occurring in the Legacy Lava DotLiquid engine as far back as v13.8.  Since DotLiquid has been deprecated and will no longer be available starting in v17.0. What are your plans on switching to the Lava Fluid engine? Here is a [link](https://community.rockrms.com/connect/ending-support-for-dotliquid-lava-engine) to more information on ending support for DotLiquid.

The Lava Fluid engine on the demo site does function correctly. The array is shuffling at random when refreshing the page and returns 'true' for the Contains filter. 
![chrome_UlRULhRncm](https://github.com/SparkDevNetwork/Rock/assets/134172019/c32b70bf-204f-4786-9f0a-5996204891f3)

**Comment by jamrphy-bwd** [comment]
@chead4 Plans to switch to Fluid? Uh... sometime before v17!  🐌🐌

Glad to know this is not affecting Fluid, which means this will just resolve itself. 👍

Anyone else running into this and needing a temporary solution on DotLiquid can probably get away with something hacky like `{% assign array = array | Join:',' | Split:',' %}` to reconvert to a List type before proceeding with other filters, like `Shuffle`.
